sklearn/multioutput.py

```diff
@@ -818,6 +855,14 @@ class ClassifierChain(MetaEstimatorMixin, ClassifierMixin, _BaseChain):
         - :term:`CV splitter`,
         - An iterable yielding (train, test) splits as arrays of indices.
 
+    chain_method : {'predict', 'predict_proba', 'predict_log_proba', \
```

Comment:
This looks close to what `_get_response_values` is doing. I am thinking that it could be nice to allow for a list of priorities:

```python
["predict_proba", "decision_function"]
```

means that you try first the proba and then the decision if the estimator does not implement `predict_proba`. This could be potentially useful if you grid-search several estimator that does not implement one of the methods.

And the nice part is that this is already managed by `_get_response_values`.

Reply:
> I am thinking that it could be nice to allow for a list of priorities:

Having a look at other functions that offer similar parameter:

* has 'auto' option that tries 'predict_proba' first, then 'decision_function':
   * [`DetCurveDisplay.from_estimator`](https://scikit-learn.org/dev/modules/generated/sklearn.metrics.DetCurveDisplay.html#sklearn.metrics.DetCurveDisplay.from_estimator)
   *  [`PrecisionRecallDisplay.from_estimator`](https://scikit-learn.org/dev/modules/generated/sklearn.metrics.PrecisionRecallDisplay.html#sklearn.metrics.PrecisionRecallDisplay.from_estimator)
   * [`RocCurveDisplay.from_estimator`](https://scikit-learn.org/dev/modules/generated/sklearn.metrics.RocCurveDisplay.html#sklearn.metrics.RocCurveDisplay.from_estimator)
* allow list of strings:
   * [make_scorer](https://scikit-learn.org/dev/modules/generated/sklearn.metrics.make_scorer.html#sklearn.metrics.make_scorer)

Maybe I have missed some? Will allow list here but I wonder (in future) if it would be interesting to amend the 'auto' options to allow list instead, offering more flexibility to the user?

Reply:
I've not found a good solution to the problem that `_BaseChain.fit` and `_BaseChain.predict` are parent to both `ClassifierChain` and `RegressorChain` but `RegressorChain` does not have a `chain_method` param.

I cannot assign `chain_method_` to an attr in `_BaseChain.fit` because `RegressorChain` does not require it, so I've ended up getting the chain method (using `_BaseChain._get_chain_method`) in `_BaseChain.fit` and `_BaseChain._get_predictions` (which gets used in all 'predict' methods). I could pass `chain_method_` to `_BaseChain._get_predictions` instead but that gets messy as well. Thoughts welcome.

Reply:
Tests are currently failing because `_get_response_values` does not properly support 'predict_log_proba'. Maybe we add could support this in another PR?